src/Microsoft.CodeQuality.Analyzers/Core/Performance/DoNotUseCountWhenAnyCanBeUsed.cs

```diff
@@ -0,0 +1,358 @@
+        private static readonly LocalizableString s_localizableMessage = new LocalizableResourceString(nameof(MicrosoftPerformanceAnalyzersResources.DoNotUseCountWhenAnyCanBeUsedMessage), MicrosoftPerformanceAnalyzersResources.ResourceManager, typeof(MicrosoftPerformanceAnalyzersResources));
+        private static readonly LocalizableString s_localizableDescription = new LocalizableResourceString(nameof(MicrosoftPerformanceAnalyzersResources.DoNotUseCountWhenAnyCanBeUsedDescription), MicrosoftPerformanceAnalyzersResources.ResourceManager, typeof(MicrosoftPerformanceAnalyzersResources));
+        private static readonly DiagnosticDescriptor s_rule = new DiagnosticDescriptor(RuleId,
+                                                                             s_localizableTitle,
+                                                                             s_localizableMessage,
+                                                                             DiagnosticCategory.Performance,
+                                                                             DiagnosticHelpers.DefaultDiagnosticSeverity,
+                                                                             isEnabledByDefault: DiagnosticHelpers.EnabledByDefaultForVsixAndNuget,
+                                                                             description: s_localizableDescription,
+                                                                             helpLinkUri: "https://docs.microsoft.com/visualstudio/code-quality/" + RuleId,
+                                                                             customTags: FxCopWellKnownDiagnosticTags.PortedFxCopRule);
+
+        /// <summary>
+        /// Returns a set of descriptors for the diagnostics that this analyzer is capable of producing.
+        /// </summary>
+        /// <value>The supported diagnostics.</value>
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(s_rule);
+
+        /// <summary>
+        /// Called once at session start to register actions in the analysis context.
+        /// </summary>
+        /// <param name="context">The context.</param>
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+
+            context.RegisterCompilationStartAction(OnCompilationStart);
+        }
+
+        /// <summary>
+        /// Called on compilation start.
+        /// </summary>
+        /// <param name="context">The context.</param>
+        private static void OnCompilationStart(CompilationStartAnalysisContext context)
+        {
+            var enumerableType = context.Compilation.GetTypeByMetadataName(EnumerableMetadataName);
```

Comment:
Please add to https://github.com/dotnet/roslyn-analyzers/blob/master/src/Utilities/Compiler/WellKnownTypes.cs and https://github.com/dotnet/roslyn-analyzers/blob/master/src/Utilities/Compiler/WellKnownTypeNames.cs